parlai/mturk/core/react_server/dev/app.jsx

```diff
@@ -308,7 +312,35 @@ class StaticApp extends React.Component {
   }
 
   onValidData(valid, response_data) {
-    this.setState({task_done: valid, response_data: response_data});
+    let all_response_data = this.state.response_data;
+    let show_next_task_button = false;
+    let task_done = true;
+    all_response_data[this.state.current_subtask_index] = response_data;
+    if (this.state.current_subtask_index < this.state.num_subtasks - 1) {
+      show_next_task_button = true;
+      task_done = false;
+    }
+    this.setState(
+      {
+        show_next_task_button: show_next_task_button,
+        subtask_done: valid,
+        task_done: task_done,
+        response_data: all_response_data,
+      },
+      this.render
```

Comment:
Is this line necessary? setState should cause an implied re-render when the state changes